dipy/io/tests/test_utils.py

```diff
@@ -51,6 +52,73 @@ def test_decfa():
     assert np.all(data_rt[0, 0, 0] == np.array([25, 0, 0]))
 
 
+def is_affine_valid(affine):
+    return is_reference_info_valid(affine, [1, 1, 1], [1.0, 1.0, 1.0],
+                                   'RAS')
+
+
+def is_dimensions_valid(dimensions):
+    return is_reference_info_valid(np.eye(4), dimensions, [1.0, 1.0, 1.0],
+                                   'RAS')
+
+
+def is_voxel_sizes_valid(voxel_sizes):
+    return is_reference_info_valid(np.eye(4), [1, 1, 1], voxel_sizes,
+                                   'RAS')
+
+
+def is_voxel_order_valid(voxel_order):
+    return is_reference_info_valid(np.eye(4), [1, 1, 1], [1.0, 1.0, 1.0],
+                                   voxel_order)
+
+
+def test_reference_info_validity():
+    if is_affine_valid(np.eye(3)):
```

Comment:
How about this instead?

```
assert not is_affine_valid(np.eye(3))
```

And so on?

Reply:
@skoudoro Would you agree with that ? If I remember correctly I had a ton of these in test_stateful_tractogram.py, but change them all for a reason I can't remember ?

Was it you or is it all in my head ?

Reply:
No, you are right. We said that we should avoid `assert` because they do not give an informative error message. So, in your case, you should just do

`np.assert_equal( is_affine_valid(np.eye(3)), False)`

Reply:
@skoudoro :  I believe you meant `npt.assert_equal`? Probably even better to use `npt.assert_`?

    npt.assert_(not is_affine_valid(np.eye(3)))

If you need an informative message, you could also do: 
    
    assert not is_affine_valid(np.eye(3)), "This affine is not valid"

But, IMO, `npt.assert_` is the best. WDYT @skoudoro ?

Reply:
> I believe you meant npt.assert_equal?

yes

> But, IMO, npt.assert_ is the best...

Agree, and still with a message if possible 😄